### PR TITLE
EventSetup consumes migration for L1GtUtils, L1TGlobalUtil, HLTPrescaleProvider

### PR DESCRIPTION
**Pull request by wddgit**
#### PR description:

Migrate L1GtUtils, L1TGlobalUtil, HLTPrescaleProvider, and GenericTriggerEventFlag to make necessary consumes calls for the EventSetup objects they get. The clients of these classes are modified the minimal amount necessary to keep things working properly with those changes, but this does not include the full consumes migration of all those client modules. That will need to be done in future PRs.

There is some choice for L1GtUtils and L1TGlobalUtil whether to get data in beginRun or the Event or both. It defaults to beginRun and clients that already used that choice did not need modification.

HLTPrescaleProvider needs a configuration parameter now to tell it whether to use L1GtUtils or L1TGlobalUtil. We do not want it prefetching for both when EventSetup prefetching is enabled. The era mechanism will be used to control this choice unless someone manually sets the parameter.

This could easily break code outside the repository. If they use one of the 4 migrated classes and the constructor argument defaults are not what is required.

#### PR validation:

There is no new functionality here. All results should be the same. I just ran the full runTheMatrix tests.

### CODE REVIEW

**Comment by wddgit** [comment]
please test

**Comment by wddgit** [comment]
I checked the potential conflicts reported above and there is no real conflict. The lines of code modified are not the same and there is plenty of separation.

**Comment by wddgit** [comment]
please test

Fixed the unit test by adding the era into the PatAlgos unit tests, I used the lowest level era required but if a PAT expert has a suggestion for a higher level era that is better please let me know.

**Comment by jfernan2** [comment]
+dqm

**Comment by wddgit** [comment]
please test

I added in a commit that completes the migration for GenericTriggerEventFlag also. Seemed easier to add it in this PR than make a separate PR. Turned out to be easier than I expected, and if separate it would have conflicted with this PR.

**Comment by jfernan2** [comment]
+dqm

**Comment by chayanit** [comment]
+1

**Comment by wddgit** [comment]
I am seeing conflicting PR's getting submitted (the ones submitted before this one don't really conflict). This seems to be a very active part of the code. It would be nice if this could get merged before I have to redo it all. (Or I will resolve the conflicts once the other ones get merged...).

**Comment by fabiocos** [comment]
@rekovic @Martin-Grunewald could you please check, as a good fraction of this PR is in your domain?

File: Calibration/IsolatedParticles/plugins/IsoTrig.cc [hunk 1]
```diff
@@ -550,7 +550,8 @@ void IsoTrig::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
   desc.addUntracked<double>("vertexCutIsol", 101.0);
   desc.addUntracked<double>("tauUnbiasCone", 1.2);
   desc.addUntracked<double>("prelimCone", 1.0);
-  descriptions.add("isoTrigHB", desc);
+  desc.add<unsigned int>("stageL1Trigger", 1);
```
Comment (Martin-Grunewald) [inline Calibration/IsolatedParticles/plugins/IsoTrig.cc]: Why is the default the old "1" rather than the new/current "2"?

Comment (makortel) [reply Calibration/IsolatedParticles/plugins/IsoTrig.cc]: The default for era modifiers (i.e. without any modifiers) corresponds run 1.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28592	f874893 
- changes look essentially technical for the reco categories
- jenkins tests pass and comparisons show no differences

**Comment by Martin-Grunewald** [comment]
+1

**Comment by tlampen** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by fabiocos** [comment]
@rekovic please check this code, as far as I can see this is just a technical migration, I plan to integrate it for today's IB unless you have comments in advance.

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**